site/content/docs/contributing/project-structure.md

```diff
@@ -9,42 +9,139 @@ menu:
-│       ├── get/     # List kubeconfigs and clusters
-│       └── kind.go  # Root command
-├── main.go          # Entrypoint
-```
+## Code Layout
+The kind project is composed of two parts: the CLI and the packages that        
+implement kind's functionality.
+We will go more in depth below.
+
+### CLI
 
-The CLI is built using [cobra][cobra] and you can see the app's entrypoint, [`main.go`][main.go], at the root level of the repository.
-The CLI commands can be found in the directory [cmd][cmd]. Here, you will find
-the root command [kind.go][kind.go] where we register other commands to build
-images; create, delete, and list clusters; list kubeconfig files for cluster;
-and setup logging.
+kind's CLI commands are defined in [cmd/kind](/cmd/kind).
+Each subdirectory corresponds to a command, and each of their subdirectories
+implements a subcommand.
+The CLI is built using [cobra][cobra] and you can see the app's entrypoint,
 
-## Packages
+### Packages
 ```
 ├── pkg
-│   ├── build     # Build and manage images
-│   ├── cluster   # Build and manage clusters
-│   ├── docker    # Interact with Docker
-│   ├── exec      # Execute commands
-│   ├── fs        # Interact with the host file system
-│   ├── kustomize # Work with embedded kustomize commands
-│   ├── log       # Logging
+│   ├── build      # Build and manage images
+│   ├── cluster    # Build and manage clusters
+│   ├── container  # Interact with the host's container runtime
+│   ├── exec       # Execute commands
+│   ├── fs         # Interact with the host file system
+│   ├── kustomize  # Work with embedded kustomize commands
+│   ├── log        # Logging
 │   └── util
 ```
 `kind` commands rely on the functionality of the [packages directory][pkg].
 Here, you will find everything needed to build container images for `kind`;
 create clusters from these images; interact with the Docker engine and file system; customize configuration files; and logging.
 
 
+## Developer Tooling
+Kind includes some tools to help developers maintain the source code compliant to Go best coding practices using tools such as Go fmt, Go lint, and Go vet. It also includes utility scripts that will generate code necessary for kind to make use of Kubernetes-style resource definitions.
+
+Tools are included in the [kind/hack](/kind/hack) directory and fall in one of two categories:
+
+* `update-*` : make changes to the source code
+* `verify-*` : verify that source code is in a good state
+
+We will proceed by describing all of the current tooling in [/kind/hack](/kind/hack).
+
+### Update
+In order to get the project’s source code into a state likely to be compilable, the script
+[update-all.sh](/kind/hack/update-all.sh) performs the following tasks:
+
+* runs update-deps.sh to get obtain all of the project’s dependencies
+* runs update-generated.sh to generate code necessary to define a configuration schema for kind, see https://kind.sigs.k8s.io/docs/user/quick-start/#configuring-your-kind-cluster
```

Comment:
Maybe explicitly call out `to generate Kubernetes API code` here?

Reply:
done